### PR TITLE
_analyze api does not correctly use normalizers when specified

### PR DESCRIPTION
**Pull request by gaobinlong**
#48650
_analyze api does not correctly use normalizers when specified, because the custom normalizer has never been used. The unit test function-testNormalizerWithIndex() in TransportAnalyzeActionTests class passed because the test case is simple.

This commit fix the bug and use another test case in testNormalizerWithIndex.

### CODE REVIEW

**Comment by elasticcla** [comment]
Hi @gaobinlong, we have found your signature in our records, but it seems like you have signed with a different e-mail than the one used in your Git [commit](https://github.com/elastic/elasticsearch/pull/48866.patch). Can you please add both of these e-mails into your Github profile (they can be hidden), so we can match your e-mails to your Github profile?

**Comment by gaobinlong** [comment]
Hi @elasticcla , sorry, my fault, I have added the e-mail used in the commit into my Github profile. Is there someting else need to do?

File: server/src/test/java/org/elasticsearch/action/admin/indices/TransportAnalyzeActionTests.java [hunk 1]
```diff
@@ -428,13 +428,13 @@ public void testCustomCharFilterWithParameters() throws IOException {
     public void testNormalizerWithIndex() throws IOException {
         AnalyzeAction.Request request = new AnalyzeAction.Request("index");
         request.normalizer("my_normalizer");
-        request.text("ABc");
+        request.text("Wi-fi");
```
Comment (cbuescher) [inline server/src/test/java/org/elasticsearch/action/admin/indices/TransportAnalyzeActionTests.java]: This is great and absolutely correct. 
It took me a while reading the code though to understand though why this fails without the additions you made in TransportAnalyzeAction. Can you add something along the lines of the following comment to explain that we expect this normalizer to use the "keyword" tokenizer and "lowercase" filter. 
The standard analyzer thats picked by error now also seems to lowercase which is why the test passed with the current test text. Lets make the assumptions around the tokenizer also very clear here.

```suggestion
        // this should be tokenized with the "keyword" tokenizer and then lowercased
        request.text("Wi-fi");
```

**Review by cbuescher — COMMENTED** [review]
Hi @gaobinlong, great catch and the fix and the test changes look great to me!
I left a short comment and a suggestion around explaining the assumptions in the test better but other than that LGTM. I will wait for you to come back before running some final tests before merging.

File: server/src/test/java/org/elasticsearch/action/admin/indices/TransportAnalyzeActionTests.java (hunk 1 above)
Comment (gaobinlong) [reply server/src/test/java/org/elasticsearch/action/admin/indices/TransportAnalyzeActionTests.java]: Hi @cbuescher , from the document of [Normalizer](https://www.elastic.co/guide/en/elasticsearch/reference/current/analysis-normalizers.html) , I notice that normalizer do not have a tokenizer and they are only worked on keyword datatype. With the standard analyzer, the text "Wi-fi" will emit "wi" and "fi",  so I think the comment can be as follows:
```suggestion
        // this should be lowercased and only emit a single token
        request.text("Wi-fi");
```

Comment (cbuescher) [reply server/src/test/java/org/elasticsearch/action/admin/indices/TransportAnalyzeActionTests.java]: Sure, lets go with that. I'll run tests and merge after you commited this version. Thanks a lot.

**Review by cbuescher — APPROVED** [review]
Thanks for the update, LGTM. I will run our CI tests before merging.

**Comment by cbuescher** [comment]
@elasticmachine test this please

**Comment by cbuescher** [comment]
@elasticmachine update branch

**Comment by cbuescher** [comment]
@elasticmachine test this please

**Comment by cbuescher** [comment]
Thanks again @gaobinlong for working on this bug, the PR was much appreciated. I merged it to master and the 7.x branch, so the fix will be included in the upcoming 7.6 release.

**Pull request merged**